packages/uni_app/lib/view/lazy_consumer.dart

```diff
@@ -28,13 +28,17 @@ class LazyConsumer<T1 extends StateProviderNotifier<T2>, T2>
     required this.hasContent,
     required this.onNullContent,
     this.contentLoadingWidget,
+    this.mapper,
     super.key,
   });
 
   final Widget Function(BuildContext, T2) builder;
   final bool Function(T2) hasContent;
   final Widget onNullContent;
   final Widget? contentLoadingWidget;
+  final T2 Function(T2)? mapper;
+
+  static T2 _defaultMapper<T2>(T2 value) => value;
```

Comment:
Ideally the output type could be different from the  input type

Reply:
I misplaced the comment, it should be next to the mapper property!

Reply:
Please check the new changes, I'm not sure if I did it the correct way.

Reply:
Okay, I tried to do what I was thinking of and I couldn't do it. I think we can leave it using T2, as it is better than dynamic... Sorry!

Reply:
Sure, already reverted the changes :+1: